### ISSUE TITLE
umask issue

### ISSUE THREAD

**Issue by mlucool** [pnpm/pnpm#3807]
### Code to reproduce the issue:

```
pnpm config set store-dir `pwd`/mystore

umask 077
mkdir foo && cd foo
pnpm init -y
pnpm install left-pad
# All files are correctly 600
ll node_modules/left-pad/

cd ..
umask 002
mkdir bar && cd bar
pnpm init -y
pnpm install left-pad
# All files are incorrectly 600, they should be 664
ll node_modules/left-pad/
```

```
pnpm config set store-dir `pwd`/mystore2

umask 002
mkdir bar && cd bar
pnpm init -y
pnpm install left-pad
# All files are correctly 664
ll node_modules/left-pad/
```

### Expected behavior:
The permissions of the files should be driven by the current umask only and not related to previous installs.

### Actual behavior:
As shown above, the first install of a module sets the permissions for all future installs. This is bad because sometimes you want to share code and sometimes you don't.

If fixing this is hard, I'd be happy to have a mode where I can set my stores umask and always use one for my store, independent of the process umask.

### Additional information:

 - `node -v` prints: v16.9.1
 - Windows, macOS, or Linux?: Linux

**Comment by zkochan** [comment]
> If fixing this is hard, I'd be happy to have a mode where I can set my stores umask and always use one for my store, independent of the process umask.

I don't think we can fix it because there is a central content-addressable store with specific permissions, and the same permissions will be applied to whichever node_modules directories the files from this store are linked to.

Adding an option to control with what mode files are written to the store should be possible.
